### PR TITLE
🌱 Add ClusterLabelName to DockerMachinePool object

### LINKED ISSUE #5340: Add cluster name field or label to DockerMachinePool
**User Story**

As a user/operator I would like to be able to easily figure out to which cluster a DockerMachinePool belongs.

**Detailed Description**

Usually we have either a `.spec.clusterName` field or the `cluster.x-k8s.io/cluster-name` label on resources belonging to a specific cluster. DockerMachinePool has neither.

**Anything else you would like to add:**

Q1: Should it be a field or a label? I'm not sure what the exact considerations are for that.
Q2: Who applies the label, the user who creates the resource or a controller? (if a controller: which one?)

/kind feature

### PR DESCRIPTION
**Pull request by anusha94**
This will allow the operator to easily identify to which `cluster` a
`DockerMachinePool` belongs

Fixes #5340

### CODE REVIEW

**Comment by anusha94** [comment]
@sbueringer I didn't find any related test file to add this assertion. Can you point me to it?
If not, I am happy to backfill some tests :)

**Comment by sbueringer** [comment]
> @sbueringer I didn't find any related test file to add this assertion. Can you point me to it? If not, I am happy to backfill some tests :)

Looks to me like we don't have any unit tests for the DockerMachinePool controller 
/ok-to-test

**Comment by sbueringer** [comment]
@fabriziopandini Just to confirm, did you mean the MachinePool or the DockerMachinePool controller in https://github.com/kubernetes-sigs/cluster-api/issues/5340#issuecomment-931526849?

I think DockerMachinePool makes sense as it's the controller for the DockerMachinePool resource and the MachinePool controller (If I saw correctly) currently doesn't access the InfraMachinePool in any way, but I'm not really sure.

**Comment by fabriziopandini** [comment]
@sbueringer I think that the MachinePool controller should add the cluster label this his external objects, so this apply both to DockerMachinePool and to any other <Infra>MachinePool implementation. Also, this behavior is consistent with what happens in the Cluster or Machine controller.

However, with the disclaimer that I don't use machinePools often and that I did not tried to reproduce the issue, it seems that the label is being already added:

https://github.com/kubernetes-sigs/cluster-api/blob/fc52fe3d5cfe05d436a7d2f0ba3cf8a1399aad59/exp/controllers/machinepool_controller_phases.go#L122-L127

🤔

**Comment by sbueringer** [comment]
@fabriziopandini That makes sense. 

@anusha94 Sorry my bad. I made an error when testing the Cluster columns. I think I just deployed a DockerMachinePool on top of the regular quickstart e2e test and forgot to enable the MachinePool feature gate.

An example from our e2e test: https://storage.googleapis.com/kubernetes-jenkins/logs/periodic-cluster-api-e2e-main/1444945386291597312/artifacts/clusters/bootstrap/resources/machine-pool-d3y9i2/DockerMachinePool/machine-pool-g7rklj-dmp-0.yaml

So tl;dr it really actually is already implemented, sorry :/

**Comment by anusha94** [comment]
Makes sense! No worries.
@sbueringer closing the PR.
